utils/zero.go

```diff
@@ -7,3 +7,13 @@ package utils
 func Zero[T any]() T {
 	return *new(T)
 }
+
+// ZeroSlice sets all values of the provided slice to the type's zero value.
+//
+// This can be useful to ensure that the garbage collector doesn't hold
+// references to values that are no longer desired.
+func ZeroSlice[T any](s []T) {
+	for i := range s {
+		s[i] = *new(T)
+	}
+}
```

Comment:
I'm not sure if we could've used this in the past (typically we zero out the slice 1 element at a time)... But it seems generally useful.